src/Bucket.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <array>
+#include <optional>
+
+template<class T, size_t N>
+class Bucket {
+    std::array<T, N> buf;
+    size_t sz;
+public:
+    Bucket() {
+        sz = 0;
+    }
+
+    void add(const T& item) {
+        if (sz < N) {

```

Comment:
nit: I don't like the silent failure here. I think the assertions should be here in the data structure and not before method invocation.